tests/utils/helper.py

```diff
@@ -0,0 +1,42 @@
+import logging
+import os
+import subprocess
+
+from ansible.parsing.dataloader import DataLoader
+from ansible.inventory.manager import InventoryManager
+
+
+def run_make_shell(args=None, tmpdir=None, **kwargs):
+    make_args = []
+    if tmpdir:
+        make_args.append('SHELL_ENV={}'.format(tmpdir))
+    if args:
+        command = args
+    else:
+        command = 'true'
+    make_args.append('C="{}"'.format(command))
+    full_command = 'make shell {args}'.format(args=' '.join(make_args))
+    logging.warning("Running: {}".format(full_command))
+    make_process = subprocess.Popen(full_command, shell=True, **kwargs)
```

Comment:
When not using shell-mode, but `execv*` mode, the command could be passed in as parts and there'd be no need for the join-by-space etc above. Shell expansion is a mess.

Reply:
But I have shell=True, not sure to understand 'When not using shell-mode, [...]'. But doc (https://docs.python.org/3/library/subprocess.html)
```
On POSIX with shell=True, the shell defaults to /bin/sh. If args is a string, the string specifies the command to execute through the shell. This means that the string must be formatted exactly as it would be when typed at the shell prompt. This includes, for example, quoting or backslash escaping filenames with spaces in them. If args is a sequence, the first item specifies the command string, and any additional items will be treated as additional arguments to the shell itself. 
```
I don't want my argument to be pass to shell itself ! I'm wrong ? (Yes shell expansion is a mess)